flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskTest.java

```diff
@@ -132,19 +139,28 @@ public void teardown() throws Exception {
 
     @Test
     public void testCleanupWhenRestoreFails() throws Exception {
-        createTaskBuilder().setInvokable(InvokableWithExceptionInRestore.class).build().run();
+        createTaskBuilder()
+                .setInvokable(InvokableWithExceptionInRestore.class)
+                .build(Executors.directExecutor())
```

Comment:
I'd rather not because we shouldn't _really_ encourage the use of direct executors. If used incorrectly (which we generally do) then the test no longer works under the same conditions as in production because the threading behavior is radically different.